runtimes/flutter/flutter.mdx

```diff
@@ -4,129 +4,398 @@ description: 'Flutter runtime for Rive.'
+);
+```
+
+**Data Binding:**
+
+```dart
+// Auto-bind with default view model instance
+final viewModelInstance = controller.dataBind(DataBind.auto());
+
+// Bind by specific instance
+final viewModelInstance = controller.dataBind(DataBind.byInstance(myInstance));
+
+// Bind by name
+final viewModelInstance = controller.dataBind(DataBind.byName("MyViewModel"));
+```
+
+### File loading
+
+The `FileLoader` class provides a unified way to load Rive files from different sources.
+
+**Loading from Assets:**
+
+```dart
+final fileLoader = FileLoader.fromAsset(
+  "assets/vehicles.riv",
+  riveFactory: Factory.rive,
+);
+```
+
+**Loading from URL:**
+
+```dart
+final fileLoader = FileLoader.fromUrl(
+  "https://example.com/animation.riv",
+  riveFactory: Factory.rive,
+);
+```
+
+**Loading from Existing File:**
+
+```dart
+final fileLoader = FileLoader.fromFile(
+  existingFile,
+  riveFactory: Factory.rive,
+);
+```
+
+Or you can load files directly using the `File` class:
+
+```dart
+// Load from asset
+final file = await File.asset("assets/vehicles.riv", riveFactory: Factory.rive);
+// Load from URL
+final file = await File.url("https://example.com/animation.riv", riveFactory:
+Factory.rive);
+// Load from path
+final file = await File.path("/path/to/animation.riv", riveFactory: Factory.rive);
+// Load from bytes
+final file = await File.decode(bytes, riveFactory: Factory.rive);
+```
+
+## Error handling
+
+The Rive Flutter package provides specific exception types for different error scenarios:
+
+- `RiveFileLoaderException`: Thrown when file loading fails
+- `RiveArtboardException`: Thrown when artboard selection fails
+- `RiveStateMachineException`: Thrown when state machine selection fails
+- `RiveDataBindException`: Thrown when data binding fails
+
+## Resource management
+
+### Manual resource management (`RiveWidget`)
+
+When using `RiveWidget` directly, you are responsible for managing all resources:
+
+```dart
+@override
+void dispose() {
+  // Dispose resources in reverse order of creation
+  viewModelInstance.dispose();
+  controller.dispose();
+  file.dispose();
+  super.dispose();
 }
 ```
 
-See subsequent runtime pages to learn how to control animation playback, state machines, and more.
+### Automatic resource management (`RiveWidgetBuilder`)
 
+When using `RiveWidgetBuilder`, the widget automatically manages most resources. You only need to dispose the file loader:
+
+```dart
+@override
+void dispose() {
+  fileLoader.dispose();
+  super.dispose();
+}
+```
 <Note>
- Using Impeller and noticing visual bugs in your app? See [our notes on rendering](/runtimes/choose-a-renderer/overview#note-on-rendering-in-flutter) with Impeller for some triaging tips
+Because the resources are managed by the `RiveWidgetBuilder`, you will not be able to access the `RiveWidgetController` (and other state) after the widget is disposed. If you need to access the controller after the widget is disposed, consider create the file and controller yourself.
```

Comment:
```suggestion
Because the resources are managed by the `RiveWidgetBuilder`, you will not be able to access the `RiveWidgetController` (and other state) after the widget is disposed. If you need to access the controller after the widget is disposed, consider creating the file and controller yourself.
```